### ISSUE TITLE
Add trashbin cleanup and file versioning

### ISSUE THREAD

**Issue by tio-trom** [YunoHost-Apps/nextcloud_ynh#612]
```
  'trashbin_retention_obligation' => 'auto, 30',
  'versions_retention_obligation' => 'auto, 30',
```

To the config file. This way we make sure that after 30 days the trashbin gets cleaned up. Also only keep file versioning for 30 days. I think it makes sense, else the server will fill up with trashed files.

**Comment by Bugsbane** [comment]
Clearing the trashbin certainly makes sense as the user has specifically chosen to delete items and then not restored them for a month. I would be hesitant to set versions to 30 days though, as this could be an unexpected behavior to users leading to data loss.

We don't want old versions filling up the server though, but I would suggest that this would be better served by setting it to *not* delete automatically by default, but suggesting this config setting in the Yunohost app description and app docs so server admins will be aware if it is set.

**Comment by tio-trom** [comment]
The file versioning means to only keep 30 versions of the same file. I think that's also reasonable.

**Comment by Bugsbane** [comment]
Ah, ok. My mistake. I didn't read closely enough assumed it was also 30 days. In that case yes, both make good sense.

**Comment by Tagadda** [comment]
If I understand correctly, `30, auto` is the same as `auto`

> `auto`: default setting. keeps files and folders in the trash bin for 30 days and automatically deletes anytime after that if space is needed (note: files may not be deleted if space is not needed).
> 
> `D, auto`: keeps files and folders in the trash bin for D+ days, delete anytime if space needed (note: files may not be deleted if space is not needed)

https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/trashbin_configuration.html

**Comment by Tagadda** [comment]
I don't think the default behavior is good for most use cases.
We could handle file retention through the config panel tho'

**Comment by lapineige** [comment]
Yeah as user won't be able to notice the change that easily, I would be more in favour of a min 30 days of retention + delete only if space needed, which serves the purpose of freeing space when it grows too much, but keep a safe month of trashbin history.

**Comment by Tagadda** [comment]
Make admins notice the change easily, that's a mission for POST_UPGRADE.md o/

**Comment by tootbrute** [comment]
For those of you who found this issue because your hard drive if full...

More info about the [trashbin options](https://docs.nextcloud.com/server/18/admin_manual/configuration_server/config_sample_php_parameters.html#deleted-items-trash-bin)

You can edit your config.php file here:
`/var/www/nextcloud/config/config.php`

Another vote for `please have a reasonable default like delete files older than 30 days.`

**Comment by tomdereub** [comment]
I also think it would be nice to have a default parameter that doesn't keep files indefinitely. I would propose : 
```
  'trashbin_retention_obligation' => 'auto, 30',
  'versions_retention_obligation' => 'auto, 90',
```
I made a PR to add doc about this. The doc will have to be updated if the default yunohost parameter is changed.

**Comment by lapineige** [comment]
I would suggest longer default, especially as it will be applied for all users with no tweak to the config file on the next update (to give them time to react), and to not surprise too much the users.

I would go for 2 months (60 days) for the trashbin (it can be suited for least-active servers), and 3 months (90 days) for versioning for safeguarding.

> I made a PR to add doc about this. The doc will have to be updated if the default yunohost parameter is changed.

Nice :+1:
An admin doc message should be displayed after the upgrade, with instructions about how to change the parameters imho.
If someone have time to contribute to it, even a config panel option so there would be no manual editing of the config file (and future conflicts) and to make it more accessible.
